### PR TITLE
[improvement](topn)add per scanner limit check for new scanner

### PR DESCRIPTION
**Pull request by xiaokang**
# Proposed changes

Optimize for key topn query like `SELECT * FROM store_sales ORDER BY ss_sold_date_sk, ss_sold_time_sk LIMIT 100` (ss_sold_date_sk, ss_sold_time_sk is prefix of table sort key). 

Check per scanner limit and set eof true to reduce the data need to be read.

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/scan/vscanner.cpp [hunk 1]
```diff
@@ -62,10 +62,18 @@ Status VScanner::get_block(RuntimeState* state, Block* block, bool* eof) {
...
                 RETURN_IF_ERROR(_filter_output_block(block));
+                // record rows return (after filter) for _limit check
+                _num_rows_return += block->rows();
             }
         } while (block->rows() == 0 && !(*eof) && raw_rows_read() < raw_rows_threshold);
     }
 
+    // set eof to true if per scanner limit is reached
+    // currently for query: ORDER BY key LIMIT n
+    if (_limit > 0 && _num_rows_return > _limit) {
```
Comment (cambyzju) [inline be/src/vec/exec/scan/vscanner.cpp:73]: why not '_num_rows_return >= _limit' ?

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.9 seconds
 load time: 642 seconds
 storage size: 17123283336 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221221074213_clickbench_pr_66107.html

File: be/src/vec/exec/scan/vscanner.cpp (hunk 1 above)
Comment (xiaokang) [reply be/src/vec/exec/scan/vscanner.cpp:73]: yes, it's more accurate using >=

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**